### PR TITLE
feat: Hosting / AWS Lambda / AWS CDK

### PR DESCRIPTION
**Pull request by aheissenberger**
This is a boilerplate for deploying your Vike app to AWS using the AWS Cloud Development Kit (CDK) inkluding creating a custom domain in Route53.

* supports serverless server: `hono`, `hattip`
* use AWS CDK for deployment
* optional can create Certificates and Route53 entries for Custom Domains
* provides tests to validate the AWS Lambda handler returning the home page of the test setup

**Example Setup:**
`pnpm cli --react --hono --aws`

check `README.md` on how to configur an AWS Account.

Deploy to AWS: `pnpm deploy:aws`

### CODE REVIEW

File: boilerplates/aws/files/cdk/bin/infrastructure.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env node
+import "source-map-support/register";
+import * as cdk from "aws-cdk-lib";
+import { VikeStack } from "../lib/vike-stack";
+
+import * as certificatemanager from "aws-cdk-lib/aws-certificatemanager";
+import * as route53 from "aws-cdk-lib/aws-route53";
+
+// eslint-ignore
```
Comment (aheissenberger) [inline boilerplates/aws/files/cdk/bin/infrastructure.ts:9]: @magne4000 Any idea on how to handle typescript type check and listing on an import `import { stackNameSuffix } from "../stack-name-suffix.json";` which is a file `$stack-name-suffix.json.ts` which gets created on template creation?

Comment (magne4000) [reply boilerplates/aws/files/cdk/bin/infrastructure.ts:9]: There is a workaround to remove the comments in the generated file at least.
Just precede the comments with `//# BATI.has("REMOVE-COMMENT") || "remove-comments-only"`:

```
//# BATI.has("REMOVE-COMMENT") || "remove-comments-only"`
// eslint-ignore
// @ts-ignore file will be generated by the transformer
import { stackNameSuffix } from "../stack-name-suffix.json";
```

Comment (magne4000) [reply boilerplates/aws/files/cdk/bin/infrastructure.ts:9]: (so no actual way to handler typescript properly in IDE there)

**Comment by magne4000** [comment]
Impressive PR!
I pushed a few minor changes. I will need to see if we should move some of the code to a `vite-plugin-cdk` or `vike-aws` in the future.

**Comment by aheissenberger** [comment]
> I will need to see if we should move some of the code to a `vite-plugin-cdk` or `vike-aws` in the future.
yes, extracting a Level 3 `vike` cdk construct would allow a wider usage

**Pull request merged**